src/components/CardEditor/CardEditForm/CardEditForm.jsx

```diff
@@ -47,10 +47,20 @@ const propTypes = {
    * getValidDataItems(card, selectedTimeRange)
    */
   getValidDataItems: PropTypes.func,
-  /** an array of dataItem string names to be included on each card
+  /** if provided, returns an array of strings which are the timeRanges to be allowed
+   * on each card
+   * getValidTimeRanges(card, selectedDataItems)
+   */
+  getValidTimeRanges: PropTypes.func,
+  /** an array of dataItems to be included on each card
    * this prop will be ignored if getValidDataItems is defined
    */
-  dataItems: PropTypes.arrayOf(PropTypes.string),
+  dataItems: PropTypes.arrayOf(
+    PropTypes.shape({
+      dataSourceId: PropTypes.string,
```

Comment:
It looks like we are using these values in the component logic but we do not have this as required or do any checks for its value. Is this correct? if so we should do one or the other.

Reply:
Actually any logic involving the `dataItems` acts properly because it defaults to an empty array when not present. I was unable to find anywhere that it would cause problems.